controllers/volsync/vshandler.go

```diff
@@ -1043,6 +1027,17 @@ func (v *VSHandler) validateSnapshotAndEnsurePVC(rdSpec ramendrv1alpha1.VolSyncR
 		}
 	}
 
+	pvc, err := v.getPVC(rdSpec.ProtectedPVC.Name)
+	if err != nil {
+		return err
+	}
+
+	// Once the PVC is restored/rolled back, need to re-add the annotations from old Primary
+	err = v.addBackOCMAnnotationsAndUpdate(pvc, rdSpec.ProtectedPVC.Annotations)
+	if err != nil {
+		return err
```

Comment:
Why do remove the annotations from existing pvc, update the pvc and add the annotations back? Can we add the annotations from the old primary to the pvc object before calling createOrUpdate?

Reply:
The is a do-not-delete OCM annotation that must be deleted.